### PR TITLE
build gtk and libadwaita from source

### PR DESCRIPTION
**Pull request by albertotirla**
this is my first attempt at making latest gtk available to zrythm, independent of the version given by the gnome runtime.  
Because gtk is evolving at such a quick pace, its main branch includes important bug fixes and features which are not yet present in any release, needless to speak about the runtime. Some of these bug fixes are accessibility related, which I consider very important and therefore should be fixed in as many apps as possible.

### CODE REVIEW

File: org.zrythm.Zrythm.json [hunk 1]
```diff
@@ -38,6 +38,31 @@
         "*.a"
     ],
     "modules": [
+                {
+            "name": "gtk",
```
Comment (vchernin) [inline org.zrythm.Zrythm.json:42]:
```suggestion
            "name": "gtk",
            "buildsystem": "meson",
```

File: org.zrythm.Zrythm.json [hunk 2]
```diff
@@ -38,6 +38,31 @@
...
+            "sources": [
+                {
+                    "type": "git",
+                    "url": "https://gitlab.gnome.org/GNOME/gtk.git",
+                    "branch": "main"
```
Comment (vchernin) [inline org.zrythm.Zrythm.json:47]:
```suggestion
                    "commit": "965b3025d97fcc3682b861e32b01c5fdcc684e31",
                    "x-checker-data": {
                        "type": "json",
                        "url": "https://gitlab.gnome.org/api/v4/projects/gnome%2Fgtk/repository/commits/main",
                        "commit-query": ".id",
                        "version-query": ".short_id",
                        "timestamp-query": ".committed_date"
                    }
```

File: org.zrythm.Zrythm.json [hunk 3]
```diff
@@ -38,6 +38,31 @@
...
+            "buildsystem": "meson",
+            "config-opts": [
+                "-Dexamples=false",
+                "-Dtests=false"
+            ],
+            "sources": [
+                {
+                    "type": "git",
+                    "url": "https://gitlab.gnome.org/GNOME/libadwaita.git",
+                    "branch": "main"
```
Comment (vchernin) [inline org.zrythm.Zrythm.json:62]:
```suggestion
                    "commit": "1c2b09bbe12737fd2b107e0816a5b4f57cfc4a16",
                    "x-checker-data": {
                        "type": "json",
                        "url": "https://gitlab.gnome.org/api/v4/projects/gnome%2Flibadwaita/repository/commits/main",
                        "commit-query": ".id",
                        "version-query": ".short_id",
                        "timestamp-query": ".committed_date"
                    }
```

**Review by vchernin — CHANGES_REQUESTED** [review]

**Comment by vchernin** [comment]
Thanks for the PR, suggested are just some minor fixes.

**Review by vchernin — APPROVED** [review]

**Comment by alex-tee** [comment]
Why is xmlb needed? Also isn't appstream included in the runtime?

**Pull request merged**